packages/storage/src/providers/s3/types/options.ts

```diff
@@ -224,6 +226,12 @@ export type UploadDataOptions = CommonOptions &
 		 * @default false
 		 */
 		preventOverwrite?: boolean;
+		/**
+		 * Indicates the algorithm used to create the checksum for the object.
+		 * This checksum can be used as a data integrity check to verify that the data received is the same data that was originally sent.
+		 * @default undefined
```

Comment:
To build on what Allan suggested, I'd make it even less verbose:

```suggestion
		/**
		 * The algorithm used to compute a checksum for the object. Used to verify that the data received by S3 
		 * matches what was originally sent. Disabled by default.
		 *
		 * @default undefined
```